libcalico-go/lib/epstatusfile/status_file_watcher_test.go

```diff
@@ -65,6 +70,24 @@ func (e *eventRecorder) OnFileDeletion(fileName string) {
 	e.OnFileEvent(fileName, "delete")
 }
 
+func (e *eventRecorder) OnInSync(s bool) {
+	e.Lock()
+	defer e.Unlock()
+	e.inSyncEvents = append(e.inSyncEvents, s)
+}
+
+func (e *eventRecorder) Events() map[string][]string {
+	e.Lock()
+	defer e.Unlock()
+	return e.fileNameToEvents
```

Comment:
You release the lock before the map is accessed in the caller.  Easiest option is to clone it
```suggestion
	return maps.Clone(e.fileNameToEvents)
```

Reply:
d'oh!